build/assets/scripts/pre-boot-tuning.sh

```diff
@@ -2,114 +2,26 @@
-        cpu_affinity+=" $core"
-    done
-    echo "CPU Affinity set to $cpu_affinity"
-}
-
-# TODO - find a more robust approach than keeping the last timestamp
-RHCOS_OSTREE_PATH=$(ls -td /boot/ostree/*/ | head -1)
-RHCOS_OSTREE_BOOTLOADER_PATH=${RHCOS_OSTREE_PATH#"/boot"}
-INITRD_GENERATION_DIR="/root/initrd"
-INITRD_NEW_IMAGE="${RHCOS_OSTREE_PATH}/iso_initrd.img"
-
-OSTREE_VERSION=$(rpm-ostree status -b | grep Version | awk '{print $2}')
-CURRENT_ENTRY_FILE=$(grep -Rls ${OSTREE_VERSION} /boot/loader/entries/)
-
-if [ -f ${INITRD_NEW_IMAGE} ] && grep -qs "iso_initrd.img" ${CURRENT_ENTRY_FILE}; then
+if [ -f /etc/sysconfig/irqbalance ] && [ -f ${SYSTEM_CONFIG_CUSTOM_FILE} ] && grep -ls "IRQBALANCE_BANNED_CPUS=${NON_ISOLATED_CPU_MASK_INVERT}" /etc/sysconfig/irqbalance; then
     echo "Pre boot tuning configuration already applied"
     echo "Setting kernel rcuo* threads to the housekeeping cpus"
-    get_cpu_mask 1
     pgrep rcuo* | while read line; do taskset -pc ${NON_ISOLATED_CPUS} $line || true; done
 else
-    # Clean up
-    rm -rf ${INITRD_GENERATION_DIR}
-    
-    # Create initrd image
-    mkdir ${INITRD_GENERATION_DIR}
-    mkdir -p ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/
-    mkdir -p ${INITRD_GENERATION_DIR}/etc/systemd/
-    mkdir -p ${INITRD_GENERATION_DIR}/etc/sysconfig/
-    touch ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
-    touch ${INITRD_GENERATION_DIR}/etc/sysconfig/irqbalance
-    touch ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
-    chmod +x ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
-
-    get_cpu_mask 1
-    echo '#!/bin/sh
+    #Set IRQ balance banned cpus
+    if [ ! -f /etc/sysconfig/irqbalance ]; then
+        touch /etc/sysconfig/irqbalance
+    fi
 
-    type getargs >/dev/null 2>&1 || . /lib/dracut-lib.sh
-
-    #cpumask="$(getargs non_iso_cpumask)"
-    cpumask='$non_iso_cpumask'
-
-    log()
-    {
-    echo "tuned: $@" >> /dev/kmsg
-    }
-
-    if [ -n "$cpumask" ]; then
-    for file in /sys/devices/virtual/workqueue/cpumask /sys/bus/workqueue/devices/writeback/cpumask; do
-        log "setting $file CPU mask to $cpumask"
-        if ! echo $cpumask > $file 2>/dev/null; then
-        log "ERROR: could not write CPU mask for $file"
-        fi
-    done
-    fi' > ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
-
-    # Set CPU affinity according to NON_ISOLATED_CPUS
-    get_cpu_affinity
-    echo "[Manager]" >> ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
-    echo "CPUAffinity=$cpu_affinity" >> ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
-
-    # Set IRQ banned cpu according to NON_ISOLATED_CPUS
-    get_cpu_mask 0
-    echo "IRQBALANCE_BANNED_CPUS=$non_iso_cpumask" >> ${INITRD_GENERATION_DIR}/etc/sysconfig/irqbalance
-    
-    cd ${INITRD_GENERATION_DIR}
-    find . | cpio -co >${INITRD_NEW_IMAGE}
+    if grep -ls "IRQBALANCE_BANNED_CPUS=" /etc/sysconfig/irqbalance; then
```

Comment:
@cynepco3hahue  that is right, but if that files will hold other values then we will kill these values (i guess almost 99% this will not be the case, but still covering this 1%)

Reply:
makes sense